plugins/search/src/components/DefaultResultListItem/index.ts

```diff
@@ -1,17 +0,0 @@
-/*
- * Copyright 2021 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-export { DefaultResultListItem } from './DefaultResultListItem';
```

Comment:
How come these were deleted? O_o

Reply:
They had no public export, so the index file would have turned out empty.

I guess this is part of the problem I'm illustrating. If there was like a `publicComponents` and an `internalComponents`, then they could both have star exports in all of their index files, but the root index only exporting the former.

Reply:
If there's a components directory where 50% of its symbols are part of the public api surface, 20% are only part of the api surface by ways of being provided in the plugin ts file, and 30% are for internal use - then the exports structure ADR becomes harder to apply. So maybe there's a case for some directory splitting here

Reply:
But this file can still exist to make direct imports of the component simpler, and the ADR is fine with that too afaik? Just gotta avoid having re-exports of those components from './components/index.ts'. Tbh what I think I would like to change here is the existence of `./components/index.ts` at all, it seems like removing that would solve a fair amount of headaches and it would only require a slight modification of the ADR. Same thing likely goes for `./lib` etc.

The current guarantee is that if you start at any file you should be able to trace exports back up through parent `index.ts`s and if the chain is broken at any point then you know it's not exported. We'd need to change that so that you'd always need to check the entire chain of `index.ts` files rather than stopping at the first missing re-export

Reply:
Alright, so we'd choose one point in the chain where we stop the star propagation. That's basically what the repo already had before this PR; the root index file was the one thst did the cherry picking. That's fine as such but the drawback was that as you add things deep down in leaf directories, it's easy to forget to also add them way up at the root.

Reply:
Right, but with the addition that `./components/index.ts` is gone. That kind of automatically means that you have to manage things from the root

Reply:
Aight, will update accordingly at some later point!